addons/live-featured-project/script.js

```diff
@@ -0,0 +1,106 @@
+export default async function (addon) {
+  const showMenu = addon.settings.get("showMenu");
+  const forceAlternative = addon.settings.get("forceAlternative");
+  const alternativePlayer = addon.settings.get("alternativePlayer");
+  const autoPlay = addon.settings.get("autoPlay");
+
+  const stageElement = document.querySelector(".stage");
+  const projectId = window.Scratch.INIT_DATA.PROFILE.featuredProject.id;
+
+  // Create and append elements
+
+  const iframeElement = document.createElement("iframe");
+  iframeElement.setAttribute("allowtransparency", "");
+  iframeElement.setAttribute("width", "282");
+  iframeElement.setAttribute("height", "210");
+  iframeElement.setAttribute("frameborder", "0");
+  iframeElement.setAttribute("allowfullscreen", "");
+
+  const wrapperElement = document.createElement("div");
+  wrapperElement.id = "lfp-embed";
+
+  const changeFeaturedElement = document.createElement("div");
+  changeFeaturedElement.id = "lfp-change-featured";
+  changeFeaturedElement.textContent = "Change featured project";
+  changeFeaturedElement.addEventListener("click", () => {
+    document.querySelector('#profile-box .player [data-control="edit"]').click();
+  });
+
+  wrapperElement.appendChild(iframeElement);
+  if (document.querySelector('#profile-box .player [data-control="edit"]'))
+    wrapperElement.appendChild(changeFeaturedElement);
+  stageElement.appendChild(wrapperElement);
+
+  // Functions for loading embeds
+
+  const loadScratch = () => {
+    wrapperElement.className = "lfp-scratch";
+
+    if (showMenu) {
+      wrapperElement.classList.add("lfp-show-menu");
+    } else {
+      iframeElement.setAttribute("height", "260");
+      wrapperElement.classList.add("lfp-hide-menu");
+    }
+
+    iframeElement.setAttribute("src", `https://scratch.mit.edu/projects/embed/${projectId}/?autostart=true`);
+
+    // Auto-start Scratch players (sadly need to be done automatically)
+
+    if (autoPlay)
+      iframeElement.addEventListener(
+        "load",
+        function callback() {
+          const observer = new MutationObserver(() => {
+            if (iframeElement.contentDocument.querySelector(".loader_fullscreen_29EhP") === null) {
+              iframeElement.contentDocument.querySelector(".green-flag_green-flag_1kiAo").click();
```

Comment:
I think you could use `addon.tab.waitForElement` here, instead of your own mutation observer.

Reply:
I am observing the iframe document, not the parent document. I believe it is implemented to only observe the parent document.